File: javascript/hamming/hamming.js
export const compute = (strand1, strand2) => {
    if (strand1.length === 0 && strand2.length === 0) {
        return 0;
    } else if (strand1.length === 0) {
        throw Error('left strand must not be empty');
    } else if (strand2.length === 0) {
        throw Error('right strand must not be empty');
    } else if (strand1.length !== strand2.length) {
        throw Error('left and right strands must be of equal length');
    }
    let counter = 0;

    for (let i = 0; i < strand1.length; i++) {
        if (strand1[i] !== strand2[i]) {
            counter++;
        }
    }
    return counter;
};

File: java/hamming/src/main/java/Hamming.java
import java.util.stream.IntStream;

class Hamming {

    private String left;
    private String right;

    Hamming(String leftStrand, String rightStrand) {

        if (leftStrand.isEmpty() && rightStrand.length() > 0) {
            throw new IllegalArgumentException("left strand must not be empty.");
        }

        if (leftStrand.length() > 0 && rightStrand.isEmpty()) {
            throw new IllegalArgumentException("right strand must not be empty.");
        }

        if (leftStrand.length() != rightStrand.length()) {
            throw new IllegalArgumentException("leftStrand and rightStrand must be of equal length.");
        }

        this.left = leftStrand;
        this.right = rightStrand;
    }

    long getHammingDistance() {

        return IntStream.range(0, left.length())
                .filter(idx -> left.charAt(idx) != right.charAt(idx))
                .count();
    }
}